happiness-backend/api/models/models.py

```diff
@@ -226,16 +228,24 @@ def __init__(self, **kwargs):
         """
         self.name = kwargs.get("name")
 
-    def add_users(self, new_users):
+    def invite_users(self, users_to_invite):
         """
-        Adds users to a group
-        Requires a list of usernames to add
-        Users to be added must exist and not already be in the group
+        Invites a list of usernames to join a group
+        Requires: Users to be invited must exist and not already be in the group
         """
-        for username in new_users:
+        for username in users_to_invite:
             user = User.query.filter(User.username.ilike(username)).first()
-            if user is not None and user not in self.users:
-                self.users.append(user)
+            if user is not None and user not in self.users and user not in self.invited_users:
+                self.invited_users.append(user)
+
+    def add_user(self, user_to_add):
+        """
+        Adds a user object to a group
+        Requires: User must already have been invited to the group
+        """
+        if user_to_add in self.invited_users:
+            self.invited_users.remove(user_to_add)
+            self.users.append(user_to_add)
 
     def remove_users(self, users_to_remove):
```

Comment:
Update the spec of this function to match the new functionality.

Reply:
its already updated?